src/compiler/checker.ts

```diff
@@ -24255,7 +24267,15 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                 if (isFromInferenceBlockedSource(source)) {
                     return;
                 }
-                const inference = getInferenceInfoForType(target);
+                let inference = getInferenceInfoForType(target);
+                if (
+                    !inference && (target.flags & TypeFlags.IndexedAccess) &&
+                    inferences[0].typeParameter.flags & TypeFlags.IndexedAccess &&

```

Comment:
Especially this part of the condition is a little bit ad-hoc. What I really wanted to check here was something like `inferences[0].typeParameter === target.objectType`. The goal of this check is to determine if we are currently inferring a reverse mapped type (from what I understand this is the only situation in which we infer into an indexed access and when the type variable is an indexed access etc) and if we infer into its immediate property.

However, even though both `inferences[0].typeParameter` and `target.objectType` are represented by `T[K]`... they are not the same thing. I'm still looking into what this check should actually be checking to improve its correctness.

Reply:
Oh, I think this might be about calling `getActualTypeVariable(target.objectType)` correctly. In such a case... is the `target` for this new prop-based inference here correct? Or should it be a new~ indexed access variable with the same concrete index type but with a different object type (the one retrieved from `getActualVariable`)?